src/transactions.rs

```diff
@@ -1298,7 +1296,9 @@ impl ReadTransaction {
     ///
     /// Returns `ReadTransactionStillInUse` error if a table or other object retrieved from the transaction still references this transaction
     pub fn close(self) -> Result<(), TransactionError> {
-        if Arc::strong_count(&self.transaction_guard) > 1 {
+        let cloned = self.tree.clone_transaction_guard();
```

Comment:
As an alternative, one could have added an accessor like

```rust
fn transaction_guard(&self) -> &Arc<TransactionGuard>;
```

to avoid the unnecessary clone here. The cloning itself could then still just be written as

```rust
Arc::clone(self.tree.transaction_gurard())
```

without adding another method.

Reply:
Oh ya, that's cleaner. https://github.com/cberner/redb/pull/814